frappe/core/doctype/communication/communication.js

```diff
@@ -145,6 +147,44 @@ frappe.ui.form.on("Communication", {
 		d.show();
 	},
 
+	show_move_dialog: function(frm) {
+		var lib = "frappe.email";
+		var d = new frappe.ui.Dialog ({
+			title: __("Move"),
+			fields: [{
+				"fieldtype": "Link",
+				"options": "Email Account",
+				"label": __("Email Account"),
+				"fieldname": "email_account",
+				"reqd": 1,
+				"get_query": function() {
+					return {
+						"filters": {
+							"name": ["!=", frm.doc.email_account],
+							"enable_incoming": ["=", 1]
+						}
+					}
+				}
+			}],
+			primary_action_label: __("Move"),
+			primary_action(values) {
+				d.hide();
+				frappe.call({
+					method: "frappe.email.inbox.move_email",
+					args: {
+						communication: frm.doc.name,
+						email_account: values.email_account
+					},
+					freeze: true,
+					callback: function(r) {
+						window.history.back();
```

Comment:
hm, should this be done?

Reply:
I also thought about if it would be better to navigate back or to reload the document. I sticked to this one because it's possible to move an email to an account to which the user doesn't have access, so it exits the email by navigating back. However, this doesn't prevent the user from viewing the email anyway, because he/she can just go forward again in the browser.

In the end I think it's a preference thing and I'm open for suggestions 😃